sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/LogsHelper.cs

```diff
@@ -80,6 +83,45 @@ internal static string GetMessageAndSetProperties(LogRecord logRecord, IDictiona
             return message;
         }
 
+        internal static void WriteScopeInformation(LogRecord logRecord, IDictionary<string, string> properties)
+        {
+            StringBuilder builder = null;
+            int originalFormatDepth = 1;
+            logRecord.ForEachScope(ProcessScope, properties);
+
+            void ProcessScope(LogRecordScope scope, IDictionary<string, string> properties)
+            {
+                if (scope.Scope is IEnumerable<KeyValuePair<string, object>> stateDictionary)
+                {
+                    foreach (KeyValuePair<string, object> scopeItem in scope)
+                    {
+                        if (scopeItem.Key == "{OriginalFormat}")
+                        {
+                            properties.Add($"OriginalFormatScope_{originalFormatDepth++}", Convert.ToString(scope.Scope.ToString(), CultureInfo.InvariantCulture));
+                        }
+                        else if (!properties.TryGetValue(scopeItem.Key, out _))
+                        {
+                            properties.Add(scopeItem.Key, Convert.ToString(scopeItem.Value, CultureInfo.InvariantCulture));
+                        }
+                        else
+                        {
+                            AzureMonitorExporterEventSource.Log.Write($"DuplicateScopeItem{EventLevelSuffix.Warning}", $"Found duplicate scope key - {scopeItem.Key}.");
+                        }
+                    }
+                }
+                else
```

Comment:
is this possible?

Reply:
Yes. this is possible. From the sample `using (_logger.BeginScope("Some name"))`, calls this `else` part of the code. 

`using (_logger.BeginScope("Formatted {WithValue}", 12345))` invokes the `if` part.

Reply:
That'd be handled by the SDK itself. The `LogRecordScope` has enumerator, so you can simply do foreach(var scope in logRecordScope)

Reply:
Correct! I use the enumerator on line 96 from `LogRecordScope`

Reply:
I meant this : https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Implementation/LogRecordExtensions.cs#L147-L154

Reply:
If we do only this one (foreach), we will miss out the scope info from `using (_logger.BeginScope("Some name"))`.  I don't think OTLP log exporter tracks it.

This implementation has both foreach and the normal tracking